### DISCUSSION TITLE
Unable to connect Midi Input/Output

### DISCUSSION THREAD

**Discussion by turtley12** [monocasual/giada#640] [Q&A] [answered]
Hi,

I just came across this project, and it looks like it will do exactly what I wanted.

I seem to be unable to select a midi input or output, and I have no idea where to start troubleshooting.

I'm using Pipewire on Arch Linux. Same problem on both the Appimage, and installing from the AUR.

There are no devices listed in the port selection dropdowns, and there aren't any midi ports appearing in my patchbay.

I am fairly sure that this is a problem with giada and not the rest of the system, because I can connect my midi keyboard to other programs through pipewire without any issues.

Here's the first part of the debug messages when starting giada. 

```
[init] Giada 0.23.2
[init] Build date: Apr 15 2023
[init] Release build
[init] Dependencies:
[init]   FLTK - 1.4.0
[init]   RtAudio - 6.0.0beta1
[init]   RtMidi - 5.0.0
[init]   Libsamplerate
[init]   Libsndfile - libsndfile-1.2.0
[init]   JSON for modern C++ - 3.11.2
[init]   JUCE - 7.0.1
[KA] Compiled RtAudio APIs: 3
  JACK
  PulseAudio
  ALSA
[KM] Compiled RtMidi APIs: 2
  ALSA
  JACK
[MidiMapper::init] total midimaps found: 0
[Engine::init] MIDI map read failed!
[KA] using API 4
[KA] Opening device out=0, in=-1, samplerate=44100
[KA] 6 device(s) found
  0) Built-in Audio Analog Stereo
      ins=4 outs=2 duplex=2
      isDefaultOut=1 isDefaultIn=1
      sampleRates:
	48000 
  1) PCM2902 Audio Codec Analog Stereo
      ins=4 outs=2 duplex=2
      isDefaultOut=0 isDefaultIn=0
      sampleRates:
	48000 
  2) Firefox
      ins=2 outs=0 duplex=0
      isDefaultOut=0 isDefaultIn=0
      sampleRates:
	48000 
  3) Giada
      ins=2 outs=0 duplex=0
      isDefaultOut=0 isDefaultIn=0
      sampleRates:
	48000 
  4) Built-in Audio Analog Stereo
      ins=4 outs=2 duplex=2
      isDefaultOut=0 isDefaultIn=0
      sampleRates:
	48000 
  5) PCM2902 Audio Codec Analog Stereo
      ins=4 outs=2 duplex=2
      isDefaultOut=0 isDefaultIn=0
      sampleRates:
	48000 
[KA] JACK in use, samplerate=48000
[mixer::reset] buffers ready - maxFramesInLoop=576000, framesInBuffer=1024
[mixer::enable] enabled
[KA] Start stream - latency = 1024
[LangMapper::init] total langmaps found: 0
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
[KA] RtAudio error 1: RtApiJack::callbackEvent(): the JACK buffer size has changed ... cannot process!
```

I noticed the errors at the bottom of that log, I assumed they weren't related to this problem since midi is not mentioned, but for what it's worth they seem to continue indefinitely while giada is running.

I came across https://github.com/monocasual/giada/discussions/531. While it has similar symptoms, It seems like a different problem since giada lists JACK and ALSA as compiled RtMidi APIs, as apposed to listing none like the other issue did.

If there's anything else that would help troubleshoot this please let me know.

\- Turtley12

**Comment by turtley12** [comment, marked as answer]
It appears I needed to enable the midi in and out with little buttons I missed at first. Solved.

**Reply by gvnnz** [reply]
Thanks @Turtley12 , we probably have a UI/UX issue here: those buttons are barely visible. We will look into it, thanks!
